res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,839 @@
+
+// Scratch algorithm parameters
+MC7000.scratchParams = {
+    recordSpeed: 33.3   ,// default: 33.3
+    alpha: (1.0/10)     ,// default: (1.0/10)
+    beta: (1.0/10)/32    // default: (1.0/10)/32
+};
+
+// Sensitivity of the jog wheel (also depends on audio latency)
+// lower values make it less, higher value more sensible
+MC7000.jogParams = {
+    jogSensitivity: 30  ,// default: 30
+    maxJogValue: 3      ,// default: 3
+};
+
+/*/////////////////////////////////
+//      USER VARIABLES END       //
+/////////////////////////////////*/
+
+
+/* OTHER VARIABLES - DONT'T TOUCH EXCEPT YOU KNOW WHAT YOU DO */
+
+// Resolution of the jog wheel, set so the spinny
+// Jog LED to match exactly the movement of the Jog Wheel
+// The physical resolution seams to be around 1100
+MC7000.jogWheelTicksPerRevolution = 894;
+
+// Pitch faders up and down values (see above for user input)
+MC7000.posRateRanges = [lowest/100, low/100, middle/100, high/100, highest/100];
+MC7000.negRateRanges = [highest/100, high/100, middle/100, low/100, lowest/100];
+
+// must be "true" for Needle Search to be active 
+MC7000.needleSearchTouched = [true, true, true, true];
+
+// initial value for VINYL mode per Deck (see above for user input)
+MC7000.isVinylMode = [MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn];
+
+// initialize the "factor" function for Spinback
+MC7000.factor = [];
+
+// initialize the PAD Mode to Hot Cue and all others off when starting
+MC7000.PADModeCue = [true, true, true, true];
+MC7000.PADModeCueLoop = [false, false, false, false];
+MC7000.PADModeFlip = [false, false, false, false];
+MC7000.PADModeRoll = [false, false, false, false];
+MC7000.PADModeSavedLoop = [false, false, false, false];
+MC7000.PADModeSlicer = [false, false, false, false];
+MC7000.PADModeSlicerLoop = [false, false, false, false];
+MC7000.PADModeSampler = [false, false, false, false];
+MC7000.PADModeVelSamp = [false, false, false, false];
+MC7000.PADModePitch = [false, false, false, false];
+
+// PAD Mode Colors
+MC7000.padColor = {
+    'alloff': 0x01,         // typically not needed for PADs
+    'hotcueoff': 0x02,      // lightblue Hot Cue inactive
+    'hotcueon': 0x04,       // darkblue Hot Cue active
+    'sampleroff': 0x27,     // light pink Sampler standard colour
+    'samplerloaded': 0x38,  // dark pink Sampler loaded colour
+    'samplerplay': 0x09,    // green Sampler playing
+    'rollon': 0x10,         // BeatloopRoll active colour
+    'rolloff': 0x1B,        // BeatloopRoll off colour
+    'cueloopon': 0x0D,      // Cueloop colour for activated cue point
+    'cueloopoff': 0x1A      // Cueloop colour inactive
+};
+
+/* DECK INITIALIZATION */
+MC7000.init = function () {
+	
+    // Decks
+    MC7000.leftDeck = new MC7000.Deck(1, 3);
+    MC7000.rightDeck = new MC7000.Deck(2, 4);
+    
+    // set default Master Volume to give a little room for mixing
+    engine.setValue("[Master]", "gain", 0.85);
```

Comment:
This is not really related to the controller but rather just a way to script mixxx for your specific use case. I would appreciate if you could get rid of it.

Reply:
Are you ok if I just uncomment this? like
```
    // set default Master Volume to give a little room for mixing
    // engine.setValue("[Master]", "gain", 0.85);
```

Reply:
I get where you are coming from, but merging commented code is generally discouraged as it just bloats the file. technically, you could maintain a git patch that applies the changes you need to the master but that involves quite some git foo so idk if thats an option for you.

Reply:
Just use `git add -p` to only stage the parts you want to commit.

Reply:
@Swiftb0y  Anyhow, I gonna keep it inside for now, commented out so I can easily bring it back. This is my productive mapping and I need some things for my podcasts and radio show. 
@Holzhaus clang-format is installed. I scratch my head why it's not called from the pre-commit command

Reply:
We're using `git clang-format`, not plain `clang-format`. Maybe it's not included in the Ubuntu package. @daschuer You're also using Ubuntu, right? HOW Did you install it?

Anyway, as long as you don't change any C++ files, you can just do `export SKIP=clang-format` to skip that hook.

Reply:
After reading some howto's I could install it like 
`pip install clang-format`
and now resubmitted the files